src/main/kotlin/org/rust/cargo/runconfig/CargoRunStateBase.kt

```diff
@@ -65,14 +72,38 @@ abstract class CargoRunStateBase(
      * @param processColors if true, process ANSI escape sequences, otherwise keep escape codes in the output
      */
     fun startProcess(processColors: Boolean): ProcessHandler {
-        val commandLine = cargo().toColoredCommandLine(environment.project, prepareCommandLine())
-        LOG.debug("Executing Cargo command: `${commandLine.commandLineString}`")
-        val handler = RsProcessHandler(commandLine, processColors)
-        ProcessTerminatedListener.attach(handler) // shows exit code upon termination
-        return handler
+        val targetEnvironment = runConfiguration.targetEnvironment
+        // Fallback to non-target implementation in case of local target
+        if (targetEnvironment == null) {
+            val commandLine = cargo().toColoredCommandLine(environment.project, prepareCommandLine())
+            LOG.debug("Executing Cargo command: `${commandLine.commandLineString}`")
+            val handler = RsProcessHandler(commandLine, processColors)
+            ProcessTerminatedListener.attach(handler) // shows exit code upon termination
+            return handler
+        }
+
+        val remoteRunPatch: CargoPatch = { commandLine ->
+            if (runConfiguration.buildOnRemoteTarget && sshEnvironmentClass?.isInstance(targetEnvironment) == true) {
```

Comment:
Maybe we could use a more straightforward validation here instead of Kotlin reflection:
```Kotlin
targetEnvironment.typeId == "ssh/sftp"
```
WDYT?